packages/devtools_app/test/support/mocks.dart

```diff
@@ -15,8 +19,269 @@ class MockServiceManager extends Mock implements ServiceConnectionManager {
 
   @override
   final ConnectedApp connectedApp = MockConnectedApp();
+
+  @override
+  final FakeServiceExtensionManager serviceExtensionManager =
+      FakeServiceExtensionManager();
 }
 
 class MockVmService extends Mock implements VmServiceWrapper {}
 
 class MockConnectedApp extends Mock implements ConnectedApp {}
+
+/// Fake that simplifies writing UI tests that depend on the
+/// ServiceExtensionManager.
+class FakeServiceExtensionManager extends Fake
+    implements ServiceExtensionManager {
+  bool _firstFrameEventReceived = false;
+
+  final Map<String, StreamController<bool>> _serviceExtensionController =
```

Comment:
Can we remove unnecessary types in all these class members?

Reply:
done. Also cleaned up the ServiceExtensionManager where a lot of this code was copy and pasted from to generate the fake implementation.